sycl/source/detail/memory_manager.cpp

```diff
@@ -840,10 +840,11 @@ void MemoryManager::copy_usm(const void *SrcMem, QueueImplPtr SrcQueue,
                              size_t Len, void *DstMem,
                              std::vector<RT::PiEvent> DepEvents,
                              RT::PiEvent *OutEvent) {
-  sycl::context Context = SrcQueue->get_context();
```

Comment:
are we not only removing the host device but also removing `sycl::malloc_host` for USM?  Last I checked that's still in the spec.

And if it is, then it seems like the USM operations should work, even for the host queue, no?    The spec says you need to be able to use these routines (like queue.memcpy ) to copy between just plain old malloc data and USM, and if the USM was allocated with `malloc_host` then it seems like the host queue is where this operation will end up, no?

My apologies if I'm overlooking something.

Reply:
`sycl::malloc_host` requires the user to specify a context or queue, which after this should no longer be able to be on a host device. When called, the allocation goes through the corresponding PI backends (piextUSMHostAlloc) and as such it is the responsibility of these backends to handle other USM operations too, like copy or memfill. As such the new assert should only happen if we add something internal to the implementation that needs to allocate and do other similar operations on the host device, which should be discouraged.

Reply:
Malloc_host is about allocating device-accessible memory that resides on the host. It has nothing to do with the host device.